pkg/machinery/config/encoder/encoder.go

```diff
@@ -64,7 +64,7 @@ func isSet(value reflect.Value) bool {
 	//nolint:exhaustive
 	switch value.Kind() {
 	case reflect.Ptr:
-		return !value.IsNil() && !value.Elem().IsZero()
+		return !value.IsNil()
```

Comment:
@Unix4ever this is the only fix I had to do to make new encoder backwards compatible with the default yaml marshaller, it doesn't seem to affect examples/comments in any bad way